prompts/templates.go

```diff
@@ -0,0 +1,93 @@
+package prompts
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/samber/lo"
+)
+
+var (
+	// ErrInvalidTemplateFormat is the error when the template format is invalid and
+	// not supported
+	ErrInvalidTemplateFormat = errors.New("Invalid template format")
+)
+
+// TemplateFormat is the format of the template, options: go-template, f-string, jinja2
+type TemplateFormat string
+
+const (
+	// TemplateFormatGoTemplate is the format for Go Template
+	// TemplateFormatGoTemplate
+	TemplateFormatGoTemplate TemplateFormat = "go-template"
+	// TemplateFormatFString is the format for f-string
+	// TODO: Add support for f-string
+	TemplateFormatFString TemplateFormat = "f-string"
```

Comment:
I have defined the types and moved the refactored `f-string` interpolator implementation from `exp/prompts`. I will add the support in a later on PR.